From e9e35b0e5c3ba17139ac36aa94780403c0c7c007 Mon Sep 17 00:00:00 2001
From: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
Date: Mon, 12 Aug 2024 20:32:44 -0400
Subject: [PATCH 1/2] thunderbird: remove Version = 2 from profiles

This fixes app startup for thunderbird installations that do not use
`wrapThunderbird` `nixpkgs` wrapper that enforces
`MOZ_LEGACY_PROFILES=1`. (This is, largely, Darwin installations from
`homebrew` or a similar mechanism.)

The fix similar to the other patch for Firefox:
https://github.com/nix-community/home-manager/pull/5724

This patch removes `Version=2` unconditionally for all platforms. This
is because regardless of whether the app is installed from `nixpkgs` or
externally, it should always use v1 behavior (with a separate
`installs.ini` file available for write.) This is already enforced for
`nixpkgs` packages with the wrapper anyway, so removing this line
shouldn't affect anyone.

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
---
 modules/programs/thunderbird.nix                                 | 1 -
 .../programs/thunderbird/thunderbird-expected-profiles.ini       | 1 -
 2 files changed, 2 deletions(-)

diff --git a/modules/programs/thunderbird.nix b/modules/programs/thunderbird.nix
index c15a946ec663..41663f152d27 100644
--- a/modules/programs/thunderbird.nix
+++ b/modules/programs/thunderbird.nix
@@ -27,7 +27,6 @@ let
   profilesIni = foldl recursiveUpdate {
     General = {
       StartWithLastProfile = 1;
-      Version = 2;
     };
   } (flip map profilesWithId (profile: {
     "Profile${profile.id}" = {
diff --git a/tests/modules/programs/thunderbird/thunderbird-expected-profiles.ini b/tests/modules/programs/thunderbird/thunderbird-expected-profiles.ini
index 4191f16072c6..64ba1a5ae7d7 100644
--- a/tests/modules/programs/thunderbird/thunderbird-expected-profiles.ini
+++ b/tests/modules/programs/thunderbird/thunderbird-expected-profiles.ini
@@ -1,6 +1,5 @@
 [General]
 StartWithLastProfile=1
-Version=2
 
 [Profile0]
 Default=1

From ee5cd8bfa6fc9a11dd1048e3e086f63c2b7a0b37 Mon Sep 17 00:00:00 2001
From: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
Date: Mon, 2 Sep 2024 18:06:25 -0400
Subject: [PATCH 2/2] thunderbird: deprecate darwinSetupWarning

Now that `profiles.ini` doesn't list `Version=2`, setting the
environment variables is no longer needed: implicit version is v1, which
means `[Install]` sections are stored in a separate `installs.ini` file
that is managed by Thunderbird itself, which `profiles.ini` stays
read-only.

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>
---
 modules/programs/thunderbird.nix | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/modules/programs/thunderbird.nix b/modules/programs/thunderbird.nix
index 41663f152d27..ce0894ec9730 100644
--- a/modules/programs/thunderbird.nix
+++ b/modules/programs/thunderbird.nix
@@ -231,7 +231,7 @@ in {
 
       darwinSetupWarning = mkOption {
         type = types.bool;
-        default = true;
+        default = false;
         example = false;
         visible = isDarwin;
         readOnly = !isDarwin;
@@ -324,17 +324,8 @@ in {
     ];
 
     warnings = optional (isDarwin && cfg.darwinSetupWarning) ''
-      Thunderbird packages are not yet supported on Darwin. You can still use
-      this module to manage your accounts and profiles by setting
-      'programs.thunderbird.package' to a dummy value, for example using
-      'pkgs.runCommand'.
-
-      Note that this module requires you to set the following environment
-      variables when using an installation of Thunderbird that is not provided
-      by Nix:
-
-          export MOZ_LEGACY_PROFILES=1
-          export MOZ_ALLOW_DOWNGRADE=1
+      Using programs.thunderbird.darwinSetupWarning is deprecated. The module
+      is now compatible with all thunderbird installations.
     '';
 
     home.packages = [ cfg.package ]
